File: Single-Linked-List/SinglyLinkedListCountNodesRecursive.java
// File Name: SinglyLinkedListCountNodesRecursive.java
// Program to count the number of nodes in a singly linked list using recursion

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class SinglyLinkedListCountNodesRecursive {
    static Node head = null;

    // Recursive function to count nodes
    static int count_nodes(Node temp) {
        if (temp == null)
            return 0;
        return 1 + count_nodes(temp.next);
    }

    public static void main(String[] args) {
        System.out.println("Number of nodes = " + count_nodes(head));
    }
}

File: Single-Linked-List/singly_linked_list_count_nodes_recursive.c
// File Name: singly_linked_list_count_nodes_recursive.c
// Program to count the number of nodes in a singly linked list using recursion

#include <stdio.h>
#include <stdlib.h>

// Definition of node
struct node {
    int data;
    struct node *next;
};

struct node *head = NULL;

// Recursive function to count nodes
int count_nodes(struct node *temp) {
    if (temp == NULL)
        return 0;
    return 1 + count_nodes(temp->next);
}

int main() {
    printf("Number of nodes = %d\n", count_nodes(head));
    return 0;
}
